tests/unit/customizations/test_globalargs.py

```diff
@@ -199,3 +199,34 @@ def test_cli_read_timeout(self):
         session = mock.Mock()
         globalargs.resolve_cli_read_timeout(parsed_args, session)
         self.assertEqual(parsed_args.read_timeout, 60)
+
+    def test_cli_connect_timeout(self):
+        parsed_args = FakeParsedArgs(connect_timeout='60')
+        session = mock.Mock()
+        globalargs.resolve_cli_connect_timeout(parsed_args, session)
+        self.assertEqual(parsed_args.connect_timeout, 60)
+
+    def test_cli_read_timeout_for_blocking(self):
+        parsed_args = FakeParsedArgs(no_read_timeout=True)
+        session = mock.Mock()
+        globalargs.resolve_cli_read_timeout(parsed_args, session)
+        self.assertEqual(parsed_args.read_timeout, None)
+
+    def test_cli_connect_timeout_for_blocking(self):
+        parsed_args = FakeParsedArgs(no_connect_timeout=True)
+        session = mock.Mock()
+        globalargs.resolve_cli_connect_timeout(parsed_args, session)
+        self.assertEqual(parsed_args.connect_timeout, None)
+
+    def test_cli_read_timeout_for_blocking_overrides_read_timeout(self):
```

Comment:
It feels like to be consistent with other args in the CLI, we should have either of these two options:

a) The two args are mutex'd.
b) The last value (on the command line) wins.

Mutex'd seems more explicit to me so I have a slight preference.